### DISCUSSION TITLE
WSL Seg Fault

### DISCUSSION THREAD

**Discussion by MegaMech** [CLIUtils/CLI11#948] [Q&A]
In WSL2 Ubuntu, I get a seg fault in `_parse()`. I think I'm using C++ 20 gcc, cmake

`./CubeOS generate -m code` <-- seg faults
`./CubeOS generate` <-- works fine
`./CubeOS generate -m` <-- help text pops up as normal

Any ideas what might be going wrong?

```cpp
void BindOTRMode(CLI::App& app){
    auto otr = app.add_subcommand("generate", "Extracts the rom");
    std::string mode;
    std::string filename;
    bool otrMode = false;

    otr->add_option("-m,--mode", mode, "extract_mode modes code otr");

    otr->parse_complete_callback([&]() {
        printf("test\n");
    });
}

int main(int argc, char **argv) {
    CLI::App app{"CubeOS - Rom extractor"};

    app.require_subcommand(1);

    BindOTRMode(app);

    try {
        app.parse(argc, argv);
    } catch (const CLI::ParseError &e) {
        return app.exit(e);
    }
    return 0;
}
```

**Comment by phlptp** [comment]
mode in this case is a local variable in the BindOTRMode function.  The add_option takes a reference to it, which becomes invalid when the method returns.  Thus when your app uses the -m option, and tries to assign something to it you will get a seg fault in most cases.    The string you bind in that case needs to live beyond the parse call.  

Most of the time when I do something like this the data for a subcommand lives in a class that gets instantiated in the main loop, or something similar.

**Comment by MegaMech** [comment]
Thanks! That did it!

(I just moved everything into main)
